### PR TITLE
[XLA:GPU] Introduce platform_id_ member variable

### PR DESCRIPTION
**Pull request by whchung**
Initial step to extract commonality between `nvptx_compiler` and `amdgpu_compiler`.

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/amdgpu_compiler.h [hunk 1]
```diff
@@ -79,6 +79,8 @@ class AMDGPUCompiler : public LLVMCompiler {
   static const char* kDataLayout;
 
  private:
+  se::Platform::Id platform_id_;
```
Comment (jerryyin) [inline tensorflow/compiler/xla/service/gpu/amdgpu_compiler.h]: Does it make sense if the id is a `static const` member? Just to be safer, I feel like The `AMDGPUCompiler` should only have hte id as `kROCmPlatformId` under any circumstances.

Comment (whchung) [reply tensorflow/compiler/xla/service/gpu/amdgpu_compiler.h]: The eventual goal is to merge two subclasses so I feel the field should be PlatformId

Comment (whchung) [reply tensorflow/compiler/xla/service/gpu/amdgpu_compiler.h]: And we populate the field thru ctor

Comment (jerryyin) [reply tensorflow/compiler/xla/service/gpu/amdgpu_compiler.h]: Right... I was just looking at the code as is. When they are merged, there need a way to populate both compilers.

**Pull request merged**